vllm/entrypoints/openai/serving_engine.py

```diff
@@ -147,15 +148,26 @@ def _validate_prompt_and_tokenize(
             self,
             request: Union[ChatCompletionRequest, CompletionRequest],
             prompt: Optional[str] = None,
-            prompt_ids: Optional[List[int]] = None) -> List[int]:
+            prompt_ids: Optional[List[int]] = None,
+            truncate_input_tokens: Optional[int] = None) -> List[int]:
         if not (prompt or prompt_ids):
             raise ValueError("Either prompt or prompt_ids should be provided.")
         if (prompt and prompt_ids):
             raise ValueError(
                 "Only one of prompt or prompt_ids should be provided.")
 
-        input_ids = prompt_ids if prompt_ids is not None else self.tokenizer(
-            prompt).input_ids
+        if prompt_ids is None:
+            tokenizer_kwargs = {}
+            if truncate_input_tokens is not None:
+                tokenizer_kwargs["truncation"] = True
+                tokenizer_kwargs["max_length"] = truncate_input_tokens
```

Comment:
Suggested simplification for allocation efficiency
```suggestion
            tokenizer_kwargs = {} if truncate_input_tokens is not None else {
                "truncation": True, "max_length": truncate_input_tokens,
            }
```